ptf_tests/common/utils/test_utils.py

```diff
@@ -1017,7 +993,249 @@ def host_check_netperf(remote=False, hostname="",username="",password=""):
+    print (f"Execute cmd \"{cmd}\"")
+    
+    j=1
+    while j<3:
+        if j>2: print (f"{j} try to execute cmd \"{cmd}\"")
+        conn.sendCmd(cmd)
+        output = conn.readResult()
+        if len(output.split("\n")) <8:
+            j +=1
+        else:
+            break
+    if j>3:
+        print (f"FAIL: failed to execue \"{cmd}\" after {j} times")
+        return False
+   
+    iter_rslt = iter(output.split("\n"))
+    # Build  port counter dict
+    for each in iter_rslt:
+        each = each.strip()
+        if each.isspace() or len(each)==0:
+            continue
+        if "RX:" in each:
+            key="RX"
+            counter[key]=dict()
+            name_list = each.split(":")[1].strip().split()
+            val_list = next(iter_rslt, None).split()
+            for i in range(len(name_list)):
+                counter[key][name_list[i]]=int(val_list[i])
+        elif "RX errors" in each:
+            key="RX errors"
+            counter[key]=dict()
+            name_list = each.split(":")[1].strip().split()
+            val_list = next(iter_rslt, None).split()
+            for i in range(len(name_list)):
+                counter[key][name_list[i]]=int(val_list[i])
+        elif "TX:" in each:
+            key="TX"
+            counter[key]=dict()
+            name_list = each.split(":")[1].strip().split()
+            val_list = next(iter_rslt, None).split()
+            for i in range(len(name_list)):
+                counter[key][name_list[i]]=int(val_list[i])
+        elif "TX errors" in each:
+            key="TX errors"
+            counter[key]=dict()
+            name_list = each.split(":")[1].strip().split()
+            val_list = next(iter_rslt, None).split()
+            for i in range(len(name_list)):
+                counter[key][name_list[i]]=int(val_list[i])
+    if counter:        
+        print (f"PASS: The VM {interface} counter is built successfuly")
+        ##
+        return counter
+   
+    print (f"FAIL: unable to build {interface} counter due to output \n{output}")
+    return False
+
+def vtysh_config_frr_bgp_attr(asnumber, neighbor="",addr_family="",
+                router_id="", network="",hostname="",username="",password=""):
+    """
+    A funtion to configure frr BGP attributes
+    """
+    print (f"Configure BGP attributes")
+    if hostname and username and password:
+        remote=f"{username}@{hostname}"
+        config = pexpect.spawn("ssh " + remote) 
+        config.expect("password:")
+        config.sendline(password)
+        i = config.expect (['Permission denied', '[~#\]\$]'])
+        if i == 0:
+            print(f"Permission denied by {hostname}. Unable to login")
+            config.kill(0)
+        elif i == 1:
+            print('Connected Successfully.')
+            config.sendline("vtysh")
+            config.expect("#")
+    else:
+        config= pexpect.spawn("vtysh")
+        config.expect("#")
+    try:
+        if not asnumber:
```

Comment:
asnumber is a mandatory argument here, by "not asnumber" do you mean asnumber=0 or asnumber="" or None?